apps/daimo-mobile/src/logic/systemContacts.ts

```diff
@@ -0,0 +1,149 @@
+import { assert } from "@daimo/common";
+import * as Contacts from "expo-contacts";
+import { useEffect, useState } from "react";
+import { Alert, Linking } from "react-native";
+
+import { SystemContact } from "./daimoContacts";
+
+export interface ContactsAccess {
+  permission: Contacts.PermissionResponse | undefined;
+  ask: () => void;
+}
+
+export function useContactsPermission(): ContactsAccess {
+  const [contactsPermission, setContactsPermission] = useState<
+    Contacts.PermissionResponse | undefined
+  >();
+
+  useEffect(() => {
+    Contacts.getPermissionsAsync().then(setContactsPermission);
+  }, []);
+
+  const ask = async () => {
+    assert(!!contactsPermission, "Contacts permission not loaded");
+
+    if (!contactsPermission.granted) {
+      await requestContactsAccess(contactsPermission.canAskAgain);
+    }
+    Contacts.getPermissionsAsync().then(setContactsPermission);
+  };
+
+  return { permission: contactsPermission, ask };
+}
+
+// Take user through a two-step flow to ask for contacts access.
+// Promise resolves when user has finished the flow (either granting
+// or denying).
+function requestContactsAccess(canAskAgain: boolean) {
+  // Step 1: Ask with our own prompt, re-assuring user.
+  return new Promise<void>((resolve) => {
+    Alert.alert(
+      "Daimo Would Like to Access Your Contacts",
+      "Your contacts are accessed locally to find and pay friends with Daimo.\n\nThey are never uploaded to our servers.",
```

Comment:
👍 👍 maybe just "They are never uploaded."

first one, idk about title case... maybe `Daimo would like to access your contacts`

or something like

> **Daimo local contacts**
> Daimo would like to access your contacts to find and pay friends. Your contacts remain private and are never uploaded or shared.

Reply:
sounds good, lets use your copy. Keeping the title-case because Apple's native notifications do title case also.